### PR TITLE
Sdem/improving tests

### PR DESCRIPTION
**Pull request by GuillermoCasas**
**Description**
I made some updates to the tests with the intention of adding them to the automatic nightly runs. Please, @philbucher , can you take a look and tell me if it is possible?

**Changelog**
- Removing non-standard python imports
- Making the structure of the tests a bit tidier
- Removing some deprecation warnings (there remains one but its solution is left for later)

**Additional info**
It will be very positive to have the tests there, since the application keeps breaking from time to time due to the sporadic use.

### CODE REVIEW

File: applications/SwimmingDEMApplication/python_scripts/derivative_recovery/standard_recoverer.py
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/derivative_recovery/standard_recoverer.py:1]:
```suggestion
```

can be removed now, just a comment

File: applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py
Comment (philbucher) [inline applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py:3]: neither scipy nor numpy are part of the python standard library, hence if you use this in a test and they are not available, the test will fail

Check [this](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/CoSimulationApplication/tests/test_CoSimulationApplication.py#L7-L11) or [this](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/CoSimulationApplication/tests/co_simulation_test_factory.py#L127-L128) to see how I solve those problems, but it can be done in different ways too.

**Review by philbucher — COMMENTED** [review]
overall it looks good to me, I added the app to the CI to check if it works (note that this will require final approval from the technical committee, but it should only be a formality here)

I saw that you are using 
~~~
kratos_add_dependency(${KRATOS_SOURCE_DIR}/applications/FluidDynamicsApplication)
kratos_add_dependency(${KRATOS_SOURCE_DIR}/applications/DEMApplication)
~~~
I recommend importing these apps in your _SwimmingDEMApplication.py_ to avoid problems related to imports. This is e.g. how er do it in the TrilinosApp, which depends on the MPICore

File: applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py
Comment (GuillermoCasas) [reply applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py:3]: Sure, I can do that.

File: applications/SwimmingDEMApplication/python_scripts/derivative_recovery/standard_recoverer.py
Comment (GuillermoCasas) [reply applications/SwimmingDEMApplication/python_scripts/derivative_recovery/standard_recoverer.py:1]: I thought so, but I have not seen it removed in most places. Perhaps this could be done all at once.

File: applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py
Comment (philbucher) [reply applications/SwimmingDEMApplication/tests/tests_python_scripts/candelier_scripts/candelier.py:3]: this is how it looks like otherwise (output of the Win CI)

~~~
Running SwimmingDEMApplication tests
Traceback (most recent call last):
  File "d:\a\Kratos\Kratos\bin\Custom\applications//SwimmingDEMApplication/tests/test_SwimmingDEMApplication.py", line 11, in <module>
    import NightTests
  File "d:\a\Kratos\Kratos\bin\Custom\applications\SwimmingDEMApplication\tests\NightTests.py", line 10, in <module>
    import TestFactory as TF
  File "d:\a\Kratos\Kratos\bin\Custom\applications\SwimmingDEMApplication\tests\TestFactory.py", line 9, in <module>
    from tests_python_scripts.candelier_scripts.candelier_analysis import CandelierBenchmarkAnalysis
  File "d:\a\Kratos\Kratos\bin\Custom\applications\SwimmingDEMApplication\tests\tests_python_scripts\candelier_scripts\candelier_analysis.py", line 10, in <module>
    import candelier
  File "d:\a\Kratos\Kratos\bin\Custom\applications\SwimmingDEMApplication\tests\tests_python_scripts\candelier_scripts\candelier.py", line 2, in <module>
    import scipy.special
ModuleNotFoundError: No module named 'scipy'
d:\a\Kratos\Kratos\bin\Custom\applications//SwimmingDEMApplication/tests/test_SwimmingDEMApplication.py
~~~

**Comment by philbucher** [comment]
cool now it works!

In order to add it to the CI the official maintainers should be know => https://github.com/KratosMultiphysics/Kratos/issues/6412#issuecomment-590281954
I guess this would be you and @maceligueta?

**Comment by GuillermoCasas** [comment]
Yes, you assume corrctly.

**Review by philbucher — APPROVED** [review]
The @KratosMultiphysics/technical-committee approves adding the SwimmingDEM app to the CI

Nice work :+1:

**Pull request merged**
